src/main/java/co/rsk/federate/watcher/FederationWatcherListenerImpl.java

```diff
@@ -0,0 +1,78 @@
+package co.rsk.federate.watcher;
+
+import co.rsk.federate.BtcToRskClient;
+import co.rsk.federate.btcreleaseclient.BtcReleaseClient;
+import co.rsk.peg.federation.Federation;
+import co.rsk.peg.federation.FederationMember;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.Objects;
+import java.util.Optional;
+
+public class FederationWatcherListenerImpl implements FederationWatcherListener {
+
+    private static final Logger logger = LoggerFactory.getLogger(FederationWatcherListenerImpl.class);
+    
+    private final FederationMember federationMember;
```

Comment:
I think this is needed here, not sure how we could start the clients beforehand.

Reply:
I see this check it's been done twice. Here in the`FederationWatcherListenerImpl` and also [here](https://github.com/rsksmart/powpeg-node/blob/c5887afe30b010f9dd0a1c24855051f504d18043/src/main/java/co/rsk/federate/BtcToRskClient.java#L131) in`BtcToRskClient`

Actually, it is using the same FedMember instance set in these places:  [FedNodeRunner.startFederate](https://github.com/rsksmart/powpeg-node/blob/76580d6b28563c68112981393ee49630f8239995/src/main/java/co/rsk/federate/FedNodeRunner.java#L188) and [FedNodeRunner.configureFederateSupport](https://github.com/rsksmart/powpeg-node/blob/76580d6b28563c68112981393ee49630f8239995/src/main/java/co/rsk/federate/FedNodeRunner.java#L348)

Having said that, I think we can do what Marcos says.

Reply:
I see, since this check is already being done [here](https://github.com/rsksmart/powpeg-node/blob/c5887afe30b010f9dd0a1c24855051f504d18043/src/main/java/co/rsk/federate/BtcToRskClient.java#L131), and we calling the start method from the triggerClientChange, we can avoid having this field and this check. Does that sound good?

Reply:
Acctualy just realized it we would still need to do that check since if we just let it to be done for `btcToRskClient` we would be starting the `btcReleaseClient` even if it is not part of the new federation.